### PR TITLE
pyoxidizer 0.17 (new formula)

### PR DESCRIPTION
**Pull request by messense**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

[pyoxidizer](https://github.com/indygreg/PyOxidizer) is a modern Python application packaging and distribution tool.

### CODE REVIEW

File: Formula/pyoxidizer.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Pyoxidizer < Formula
+  desc "Modern Python application packaging and distribution tool"
+  homepage "https://github.com/indygreg/PyOxidizer"
+  url "https://github.com/indygreg/PyOxidizer/archive/refs/tags/pyoxidizer/0.17.tar.gz"
+  sha256 "9117d411b610e29dfd8d9250cd1021afb545550fd7e698b623e913c26114f013"
+  license "MPL-2.0"
+  head "https://github.com/indygreg/PyOxidizer.git", branch: "main"
+
+  depends_on "rust"
```
Comment (carlocab) [inline Formula/pyoxidizer.rb:9]: Is this needed at runtime?

Comment (messense) [reply Formula/pyoxidizer.rb:9]: I think it can be build only, but pyoxidizer does also provide a `--system-rust` argument in `init-rust-project` command.

Comment (messense) [reply Formula/pyoxidizer.rb]: Changed to build only.

File: Formula/pyoxidizer.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Pyoxidizer < Formula
+  desc "Modern Python application packaging and distribution tool"
+  homepage "https://github.com/indygreg/PyOxidizer"
+  url "https://github.com/indygreg/PyOxidizer/archive/refs/tags/pyoxidizer/0.17.tar.gz"
```
Comment (cho-m) [inline Formula/pyoxidizer.rb:4]: I think preference is shorter GitHub URL:
```suggestion
  url "https://github.com/indygreg/PyOxidizer/archive/pyoxidizer/0.17.tar.gz"
```

File: Formula/pyoxidizer.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  sha256 "9117d411b610e29dfd8d9250cd1021afb545550fd7e698b623e913c26114f013"
+  license "MPL-2.0"
+  head "https://github.com/indygreg/PyOxidizer.git", branch: "main"
+
+  depends_on "rust" => :build
+
+  def install
+    cd "pyoxidizer" do
+      system "cargo", "install", *std_cargo_args
+    end
```
Comment (cho-m) [inline Formula/pyoxidizer.rb:14]: Need to check, but I think it is now possible to do
```suggestion
    system "cargo", "install", *std_cargo_args(path: "pyoxidizer")
```

**Comment by messense** [comment]
10.14 build error reported to upstream: https://github.com/indygreg/PyOxidizer/issues/414#issuecomment-894933182

File: Formula/pyoxidizer.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+class Pyoxidizer < Formula
+  desc "Modern Python application packaging and distribution tool"
+  homepage "https://github.com/indygreg/PyOxidizer"
+  url "https://github.com/indygreg/PyOxidizer/archive/pyoxidizer/0.17.tar.gz"
+  sha256 "9117d411b610e29dfd8d9250cd1021afb545550fd7e698b623e913c26114f013"
+  license "MPL-2.0"
+  head "https://github.com/indygreg/PyOxidizer.git", branch: "main"
+
+  depends_on "rust" => :build
+  depends_on xcode: :test if MacOS.version <= :catalina
```
Comment (carlocab) [inline Formula/pyoxidizer.rb:10]: Shouldn't this be a runtime dependency too?

Comment (cho-m) [reply Formula/pyoxidizer.rb:10]: I'm only experimenting as it looks like test block is failing due to upstream issue. Don't want to force full XCode dependency right now until upstream comments.

Comment (cho-m) [reply Formula/pyoxidizer.rb:10]: Still failed on Mojave. Currently seems to only work with macOS 11 SDK, so minimum Xcode 12.2, which means minimum macOS of 10.15.4.

We could add this as a temporary runtime dependency and merge now, and remove if upstream fixes issue.

Comment (messense) [reply Formula/pyoxidizer.rb:10]: Opened a new upstream issue to track this: https://github.com/indygreg/PyOxidizer/issues/431

**Comment by cho-m** [comment]
I am fine with current state (i.e. no Mojave support, Catalina support with XCode dependency, and Big Sur support with CLT), but will give some time for any other comments.

We probably need to add a `livecheck` block due to git tag format, but I am not sure if we want to rerun CI here as Linux took a number of retries to get past CI crate.io download issue.
```rb
  livecheck do
    url :stable
    regex(%r{^pyoxidizer/v?(\d+(?:\.\d+)+)$}i)
  end
```

**Review by carlocab — APPROVED** [review]
Fine with merging as is. We can bottle Linux separately; there's no need to tie up the macOS nodes for that.

**Review by cho-m — APPROVED** [review]

**Pull request closed without merging**